debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/BinlogReader.java

```diff
@@ -116,7 +115,7 @@
     private Heartbeat heartbeat;
     private MySqlJdbcContext connectionContext;
     private final float heartbeatIntervalFactor = 0.8f;
-    private final Set<Thread> binaryLogClientThreads = Collections.synchronizedSet(new HashSet<>(2));
+    private final Map<String, Thread> binaryLogClientThreads = new ConcurrentHashMap<>(2);
```

Comment:
You could keep it as a set, using `newSetFromMap()`:

```suggestion
    private final Map<String, Thread> binaryLogClientThreads = Collections.newSetFromMap(new ConcurrentHashMap<>(3));
```

Also changed the initial size, as 2 wouldn't be enough to hold two elements actually without a resize when using the default load factor. Not that it makes any difference (I'd just call the parameterless constructor), but if we give a size explicitly, it should be large enough.

Reply:
@gunnarmorling Modified